### ISSUE TITLE
test_multiple_pvc_concurrent_creation_deletion raises KeyError

### ISSUE THREAD

**Issue by fbalak** [red-hat-storage/ocs-ci#291]
Test `test_multiple_pvc_concurrent_creation_deletion` raises an error:
```
tests/test_pvc_concurrent_deletion_creation.py::TestMultiplePvcConcurrentDeletionCreation::test_multiple_pvc_concurrent_creation_deletion ERROR

============================================================== ERRORS ==============================================================
__________ ERROR at teardown of TestMultiplePvcConcurrentDeletionCreation.test_multiple_pvc_concurrent_creation_deletion ___________

tp = <class 'KeyError'>, value = None, tb = None

    def reraise(tp, value, tb=None):
        try:
            if value is None:
                value = tp()
            if value.__traceback__ is not tb:
                raise value.with_traceback(tb)
>           raise value

venv3/lib64/python3.6/site-packages/six.py:693: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
venv3/lib64/python3.6/site-packages/six.py:693: in reraise
    raise value
venv3/lib64/python3.6/site-packages/six.py:693: in reraise
    raise value
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

    def get_status_after_execution():
        """
        Set the environment status and assign it into ENV_STATUS_PRE dictionary.
        In addition compare the dict before the execution and after using DeepDiff
    
        Raises:
             ResourceLeftoversException: In case there are leftovers in the
                environment after the execution
        """
        get_environment_status(ENV_STATUS_POST)
    
        pod_diff = DeepDiff(
            ENV_STATUS_PRE['pod'], ENV_STATUS_POST['pod']
        )
        sc_diff = DeepDiff(
            ENV_STATUS_PRE['sc'], ENV_STATUS_POST['sc']
        )
        cephfs_diff = DeepDiff(
            ENV_STATUS_PRE['cephfs'], ENV_STATUS_POST['cephfs']
        )
        cephbp_diff = DeepDiff(
            ENV_STATUS_PRE['cephbp'], ENV_STATUS_POST['cephbp']
        )
        pv_diff = DeepDiff(
            ENV_STATUS_PRE['pv'], ENV_STATUS_POST['pv']
        )
        pvc_diff = DeepDiff(
            ENV_STATUS_PRE['pvc'], ENV_STATUS_POST['pvc']
        )
        secret_diff = DeepDiff(
            ENV_STATUS_PRE['secret'], ENV_STATUS_POST['secret']
        )
        namespace_diff = DeepDiff(
            ENV_STATUS_PRE['namespace'], ENV_STATUS_POST['namespace']
        )
        diffs_dict = {
            'pods': pod_diff,
            'sc': sc_diff,
            'cephfs': cephfs_diff,
            'cephbp': cephbp_diff,
            'pvs': pv_diff,
            'pvcs': pvc_diff,
            'secret': secret_diff,
            'ns': namespace_diff,
        }
        leftover_detected = False
    
        leftovers = {'Leftovers added': [], 'Leftovers removed': []}
        for kind, kind_diff in diffs_dict.items():
            if ADDED_RESOURCE in kind_diff:
                leftovers['Leftovers added'].append({
                    kind: kind_diff[ADDED_RESOURCE][
>                       ''.join(kind_diff[ADDED_RESOURCE])
                    ]
                })
E               KeyError: 'root[0]root[1]root[2]root[3]root[4]root[5]root[6]root[7]root[8]root[9]root[10]root[11]root[12]root[13]root[14]root[15]root[16]root[17]root[18]root[19]root[20]root[21]root[22]root[23]root[24]root[25]root[26]root[27]root[28]root[29]root[30]'

utility/environment_check.py:145: KeyError

```

**Comment by RazTamir** [comment]
Fixed with https://github.com/red-hat-storage/ocs-ci/pull/292

**Comment by jilju** [comment]
@fbalak Did you hit test failure for 'test_multiple_pvc_concurrent_creation_deletion'  or it error out only in tear down part?
The [comment](https://github.com/red-hat-storage/ocs-ci/pull/292#issuecomment-505458810) shows that one PVC (out of 200 PVCs) and its associated PV remained and that was found in environment check.
I wonder why I am not hitting this issue. I executed the test case three times today and it was passing.

Output:
[test_multiple_pvc_concurrent_creation_deletion-log.txt](https://github.com/red-hat-storage/ocs-ci/files/3326660/test_multiple_pvc_concurrent_creation_deletion-log.txt)

**Comment by fbalak** [comment]
@jilju The error was triggered in teardown part but when I tried PR https://github.com/red-hat-storage/ocs-ci/pull/292#issuecomment-505509418 it failed in test. I will test it more today.

**Comment by jilju** [comment]
@fbalak 
The KeyError issue is now Fixed in #292.
The basic issue was that PVC is not moving to Bound state. Can you confirm if this is related to rook issue [3357](https://github.com/rook/rook/issues/3357)

**Comment by fbalak** [comment]
@jilju Yes, it seems like it.

**Comment by jilju** [comment]
The KeyError issue is fixed in PR #292 
Changes introduced in PR #318 fixedw the PVC Bound issue.
Closing this issue as fixed.
